src/tools/rbd_mirror/image_replayer/snapshot/Replayer.cc

```diff
@@ -542,6 +542,11 @@ void Replayer<I>::scan_local_mirror_snapshots(
     if (m_local_mirror_snap_ns.is_non_primary() &&
         m_local_mirror_snap_ns.primary_mirror_uuid !=
           m_state_builder->remote_mirror_uuid) {
+      if (m_local_mirror_snap_ns.is_orphan()) {
```

Comment:
nit: A little different approach would seem to be to add `&& !m_local_mirror_snap_ns.is_orphan()` in the `if` condition above, so we would skip this block and then extent the `else if` below to `|| m_local_mirror_snap_ns.is_orphan()` and just entered "force promoted" state there. Though I am fine with the current version too.